runtime/vm/JFRConstantPoolTypes.cpp

```diff
@@ -536,6 +536,88 @@ VM_JFRConstantPoolTypes::addPackageEntry(J9Class *clazz)
 	return index;
 }
 
+U_32
+VM_JFRConstantPoolTypes::addPackageEntry(J9Module *fromModule, J9Package *package)
+{
+	U_32 index = U_32_MAX;
+	J9PackageIDTableEntry *pkgID = NULL;
+	PackageEntry *entry = NULL;
+	PackageEntry entryBuffer = {0};
+
+	const int NAME_BUFFER_SIZE = 256;
+	U_8 nameBuffer[NAME_BUFFER_SIZE];
+	J9UTF8 *pkgNameUTF8 = (J9UTF8 *)nameBuffer;
+
+	J9ROMClass tempClass = {0};
+	J9ROMClass *queryROMClass = &tempClass;
+
+	const U_16 pkgNameLen = J9UTF8_LENGTH(package->packageName);
+
+	/* Subtrace 2 bytes for the length field */
+	if (pkgNameLen > (NAME_BUFFER_SIZE - 2)) {
+		UDATA allocationSize = sizeof(J9ROMClass) + sizeof(J9UTF8) + pkgNameLen;
+		queryROMClass = (J9ROMClass *)j9mem_allocate_memory(allocationSize, J9MEM_CATEGORY_VM);
```

Comment:
Should this memory be freed if the package entry already exists?

Reply:
Fixed.